component/site/libraries/iCalEvent.php

```diff
@@ -64,6 +64,13 @@ function iCalEvent( &$db ) {
 	function store($updateNulls=false , $overwriteCreator = false) {
 		$user = JFactory::getUser();
 
+		$jinput = JFactory::getApplication()->input;
+		$curr_task = $jinput->get('task');
+		$ical_access = $jinput->get('access');
```

Comment:
$ical_access = $jinput->getInt('access');
this ensures the data is cleaned appropriately for security reasons